Editor/Scripts/Analytics/MeshSyncObserver.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+
+using UnityEditor;
+
+namespace Unity.MeshSync.Editor.Analytics {
+
+    /// <summary>
+    /// Hacky way to restore observers to MeshSync servers in scene
+    /// </summary>
+    [InitializeOnLoad]
+    internal static class MeshSyncObserverStartUp {
+        private const string SESSION_STATE_FLAG = "meshSyncAnalyticsSetupFlag";
+        private static DateTime m_check_Start;
+        private static readonly TimeSpan INTERVAL = TimeSpan.FromSeconds(3);
+
+        private static void updateCallback(bool limitChecks) {
+            // Limit checks to once every 3 seconds
+            if ((DateTime.Now - m_check_Start < INTERVAL) && limitChecks) {
+                return;
+            }
+
+            m_check_Start = DateTime.Now;
+
+            var alreadySetup = SessionState.GetBool(SESSION_STATE_FLAG, defaultValue: false);
+
+            // If the setup was already done, we still need to do it again at least once after a domain reload
+            if (alreadySetup) {
+                EditorApplication.update -= checkWithTimeLimits;
+            }
+
+            var array = UnityEngine.Object.FindObjectsOfType<BaseMeshSync>(includeInactive: true);
+            bool success = false;
+
+            foreach (var server in array) {
+                if (server.getNumObservers == 0) {
+                    var observer = new MeshSyncObserver();
+                    server.Subscribe(observer);
+                    success = true;
+                }
+            }
+
+            if (success) {
+                SessionState.SetBool(SESSION_STATE_FLAG, value: true);
+                EditorApplication.update -= checkWithTimeLimits;
+            }
+        }
+
+        static MeshSyncObserverStartUp() {
+            m_check_Start = DateTime.Now;
+
+            EditorApplication.update -= checkWithTimeLimits;
+            EditorApplication.update += checkWithTimeLimits;
+
+            EditorApplication.playModeStateChanged -= playModeChangedCallback;
+            EditorApplication.playModeStateChanged += playModeChangedCallback;
+        }
```

Comment:
@mloccy We need to also handle adding servers by duplication or ctrl+paste. At the moment, when you duplicate the server, observers are 0. If you register the checkWithTimeLimits handler to the EditorApplication.hierarchyChanged this will work.

Reply:
I did another test: I created a prefab from the server and put it back in the scene. It works sometimes, but if you subscribe the hierachyChanged to a new calback checkWithoutTimeLimits, it works

Reply:
Could that be because these callbacks are sharing the same state variables? So one callback is affecting the others?

Reply:
The subscription to hierarchyChanged seemed to work, prefabs are behaving as expected now

Reply:
fixed